### ISSUE TITLE
Dynamic Component Example: Cannot use IDynamicElement as an element (IContainer)

### ISSUE THREAD

**Issue by thomasstevens89** [QuestPDF/QuestPDF#452]
Using version `2022.12` and following the Optimised example of the per-page totals table from the docs, the example seems to be incompatible with this version. I cannot figure out how to modify the code to use `IDynamicElement` with `Element()`.

Code:
```C#
var header = ComposeHeader(context);
var sampleFooter = ComposeFooter(context, Enumerable.Empty<OrderItem>());
var decorationHeight = header.Size.Height + sampleFooter.Size.Height;

var rows = GetItemsForPage(context, decorationHeight).ToList();
var footer = ComposeFooter(context, rows.Select(x => x.Item));

var content = context.CreateElement(container =>
{
    container.MinimalBox().Decoration(decoration =>
    {
        decoration.Before().Element(header);

        decoration.Content().Column(stack =>
        {
            foreach (var row in rows)
                stack.Item().Element(row.Element);
        });

        decoration.After().Element(footer);
    });
});
```

Error:
```
Argument 2: cannot convert from 'QuestPDF.Elements.IDynamicElement' to 'System.Action<QuestPDF.Infrastructure.IContainer>'
```

**Comment by girlpunk** [comment]
Hi, Try changing the inside of your foreach to

```csharp
stack.Item.Element(e => e.Dynamic(row.Element));
```

The `Element()` call is expecting to accept a function to tell it what to compose, in this case we can use a lambda to compose the single dynamic component.

**Comment by thomasstevens89** [comment]
Hi, thanks for the suggestion. Unfortunately that also does not work.

`error CS0411: The type arguments for method 'DynamicComponentExtensions.Dynamic<TState>(IContainer, 
IDynamicComponent<TState>)' cannot be inferred from the usage. Try specifying the type arguments explicitly.`

It's worth noting that this same issue affects the Header and Footer elements also.

**Comment by MarcinZiabek** [comment]
@thomasstevens89 I suppose that this code is taken from official examples in the documentation? If so, I need to double check if visibility modifiers are applied correctly 🤣

I will do my best to analyze your code and this problem later today / tomorrow

**Comment by thomasstevens89** [comment]
@MarcinZiabek yes it is from the documentation. The only thing I have modified from the example is to use decoration `before`, `content` & `after` instead of `header` & `footer`.

Thanks for your quick response and help on this!

**Comment by MarcinZiabek** [comment]
This is fixed in the 2022.12.1 release. Please test 😁

**Comment by thomasstevens89** [comment]
Works perfectly now thanks!
